test/Microsoft.AspNetCore.Routing.Tests/Tree/TreeRouterTest.cs

```diff
@@ -125,6 +126,58 @@ public class TreeRouterTest
             Assert.Equal(expectedRouteGroup, context.RouteData.Values["test_route_group"]);
         }
 
+        public static TheoryData<string, object[]> MatchesRoutesWithDefaultsData =>
```

Comment:
Since you mentioned about tests not present in this area, could you add a test/variation where there are parameters with constraints and also default values. Ex: `{seg1:int=1}/{seg2:int=2}`